src/debug.h

```diff
@@ -197,6 +197,31 @@ void limitDebugClass( int );
  */
 bool debug_has_error_been_observed();
 
+/**
+ * Class for capturing debugmsg.
+ * Сan also use a captureDebugmsgLambda function instead.
+ */
+class captureDebugmsg
+{
+    public:
+        captureDebugmsg();
+        // @return std::string debugmsg
+        std::string dmsg();
+        ~captureDebugmsg();
+};
+
+/**
+ * Capturing debugmsg for a lambda function
+ *  @return std::string debugmsg
+ */
+template<typename Lambda>
+std::string captureDebugmsgLambda( Lambda func )
```

Comment:
```suggestion
std::string captureDebugmsgLambda( const std::function<void()> &func )
```
If you want this to not be a template, you can change it like this, and then move its declaration and (optionally) the definition of `class captureDebugmsg` into `debug.cpp`.

I'd also change the name, because it can be used with more than lambdas.  Maybe `captureDebugmsgDuring`?

Reply:
done